common/src/main/java/earth/terrarium/olympus/client/components/map/MapWidget.java

```diff
@@ -15,32 +16,47 @@ public class MapWidget extends BaseWidget {
     private static final ResourceLocation MAP_ICONS = ResourceLocation.withDefaultNamespace("textures/map/decorations/player.png");
 
     private final int scale;
```

Comment:
make not final and have a withScale method

Reply:
should refresh map if needed, will defer map refresh to first frame and then check if thing scale has changed to try again